src/vs/editor/contrib/dropOrPasteInto/browser/copyPasteContribution.ts

```diff
@@ -26,8 +29,36 @@ registerEditorCommand(new class extends EditorCommand {
 	}
 
 	public override runEditorCommand(_accessor: ServicesAccessor | null, editor: ICodeEditor, _args: any) {
-		CopyPasteController.get(editor)?.changePasteType();
+		return CopyPasteController.get(editor)?.changePasteType();
 	}
 });
 
-registerEditorFeature(DefaultPasteProvidersFeature);
+registerEditorAction(class extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.pasteAs',
+			label: nls.localize('pasteAs', "Paste As..."),
+			alias: 'Paste As...',
+			precondition: undefined,
+			description: {
+				description: `Paste as`,
+				args: [{
+					name: 'args',
+					schema: {
+						type: 'object',
+						properties: {
+							'id': {
```

Comment:
Should this be preferredId and could it have a description?

Reply:
Added `description`. I kept `id` as the public name because it will force the the command to try pasting with that edit type instead of showing any kind of quick pick. It also matches the existing code action command arguments